### PR TITLE
yandex-music: init at 5.13.2

### PR DESCRIPTION
**Pull request by ein-shved**
## Description of changes

Introduce YandexMusic package. This package is based on the [beta](https://music.yandex.ru/download/) version of desktop application. There is another [project](https://github.com/cucumber-sp/yandex-music-linux) on github which repacks this desktop application for linux. So, this package uses that project as source and then uses hashes and urls from it to download the official archive and then the `repack.sh` to repack it for nixpkgs derivation. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ya/yandex-music/package.nix [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  ];
+
+  ymExe = fetchurl {
+    url = version_info.ym.exe_link;
+    sha256 = version_info.ym.exe_sha256;
+  };
+
+  buildPhase = ''
+    bash "./repack.sh" -o "./app" "$ymExe"
+  '';
```
Comment (azahi) [inline pkgs/by-name/ya/yandex-music/package.nix:46]:
```suggestion
  buildPhase = ''
    runHook preBuild
    bash "./repack.sh" -o "./app" "$ymExe"
    runHook postBuild
  '';
```

File: pkgs/by-name/ya/yandex-music/package.nix [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    wrapProgram "$out/bin/yandex-music" \
+      --set-default YANDEX_MUSIC_CONFIG "$CONFIG_FILE"
+
+    install -Dm644 "./app/favicon.png" "$out/share/pixmaps/yandex-music.png"
+    install -Dm644 "./app/favicon.png" "$out/share/icons/hicolor/48x48/apps/yandex-music.png"
+    install -Dm644 "./app/favicon.svg" "$out/share/icons/hicolor/scalable/apps/yandex-music.svg"
+
+    install -Dm644 "$src/templates/desktop" "$out/share/applications/yandex-music.desktop"
+  '';
```
Comment (azahi) [inline pkgs/by-name/ya/yandex-music/package.nix:69]:
```suggestion
  installPhase = ''
    runHook preInstall
    
    mkdir -p "$out/share/nodejs"
    mv app/yandex-music.asar "$out/share/nodejs"

    CONFIG_FILE="$out/share/yandex-music.conf"
    echo "TRAY_ENABLED=${if trayEnabled then "1" else "0"}" >> "$CONFIG_FILE"
    echo "ELECTRON_ARGS=\"${electronArguments}\"" >> "$CONFIG_FILE"


    install -Dm755 "$src/templates/yandex-music.sh" "$out/bin/yandex-music"
    sed -i "s|%electron_path%|${electron}/bin/electron|g" "$out/bin/yandex-music"
    sed -i "s|%asar_path%|$out/share/nodejs/yandex-music.asar|g" "$out/bin/yandex-music"

    wrapProgram "$out/bin/yandex-music" \
      --set-default YANDEX_MUSIC_CONFIG "$CONFIG_FILE"

    install -Dm644 "./app/favicon.png" "$out/share/pixmaps/yandex-music.png"
    install -Dm644 "./app/favicon.png" "$out/share/icons/hicolor/48x48/apps/yandex-music.png"
    install -Dm644 "./app/favicon.svg" "$out/share/icons/hicolor/scalable/apps/yandex-music.svg"

    install -Dm644 "$src/templates/desktop" "$out/share/applications/yandex-music.desktop"
    
    runHook postInstall
  '';
```

File: pkgs/by-name/ya/yandex-music/package.nix [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    mv app/yandex-music.asar "$out/share/nodejs"
+
+    CONFIG_FILE="$out/share/yandex-music.conf"
+    echo "TRAY_ENABLED=${if trayEnabled then "1" else "0"}" >> "$CONFIG_FILE"
+    echo "ELECTRON_ARGS=\"${electronArguments}\"" >> "$CONFIG_FILE"
+
+
+    install -Dm755 "$src/templates/yandex-music.sh" "$out/bin/yandex-music"
+    sed -i "s|%electron_path%|${electron}/bin/electron|g" "$out/bin/yandex-music"
+    sed -i "s|%asar_path%|$out/share/nodejs/yandex-music.asar|g" "$out/bin/yandex-music"
```
Comment (azahi) [inline pkgs/by-name/ya/yandex-music/package.nix:59]: Nit: better to do this with `substituteInPlace --replace-fail`

File: pkgs/by-name/ya/yandex-music/package.nix [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  meta = {
+    description = "Yandex Music - Personal recommendations, selections for any occasion and new music";
+    homepage = "https://music.yandex.ru/";
+    downloadPage = "https://music.yandex.ru/download/";
+    changelog = "https://github.com/cucumber-sp/yandex-music-linux/releases/tag/v5.13.2";
+    license = lib.licenses.unfree;
+    platforms = [
+      "x86_64-linux"
+      "aarch64-linux"
+    ];
```
Comment (azahi) [inline pkgs/by-name/ya/yandex-music/package.nix:80]:
```suggestion
    platforms = lib.platforms.linux;
```

File: pkgs/by-name/ya/yandex-music/package.nix [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+let
+  version = "5.13.2";
+  src = fetchFromGitHub {
+    owner = "cucumber-sp";
+    repo = "yandex-music-linux";
+    rev = "v${version}";
+    hash = "sha256-yQX2GEulAQ02BndpwAAwGdTYtUlNXCgsCUdmXxPwtKU=";
+  };
+  version_info = builtins.fromJSON (builtins.readFile "${src}/utility/version_info.json");
+in
```
Comment (azahi) [inline pkgs/by-name/ya/yandex-music/package.nix:26]: Nit: these can be accessed through `finalAttrs` witout depending on let.

File: pkgs/by-name/ya/yandex-music/package.nix [hunk 6]
```diff
@@ -0,0 +1,83 @@
+{
+  fetchurl,
+  stdenvNoCC,
+  lib,
+  makeWrapper,
+
```
Comment (azahi) [inline pkgs/by-name/ya/yandex-music/package.nix:6]:
```suggestion
```

File: pkgs/by-name/ya/yandex-music/package.nix [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  p7zip,
+  asar,
+  jq,
+  python3,
+  electron,
+  fetchFromGitHub,
+
```
Comment (azahi) [inline pkgs/by-name/ya/yandex-music/package.nix:13]:
```suggestion
```

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ya/yandex-music/package.nix (hunk 5 above)
Comment (ein-shved) [reply pkgs/by-name/ya/yandex-music/package.nix:26]: just moved the let to `ymExe`

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 337425` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>yandex-music</li>
  </ul>
</details>

**Review by azahi — APPROVED** [review]

File: pkgs/by-name/ya/yandex-music/package.nix [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    install -Dm644 "./app/favicon.png" "$out/share/icons/hicolor/48x48/apps/yandex-music.png"
+    install -Dm644 "./app/favicon.svg" "$out/share/icons/hicolor/scalable/apps/yandex-music.svg"
+
+    install -Dm644 "$src/templates/desktop" "$out/share/applications/yandex-music.desktop"
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Yandex Music - Personal recommendations, selections for any occasion and new music";
```
Comment (dotlambda) [inline pkgs/by-name/ya/yandex-music/package.nix:80]:
```suggestion
    description = "Personal recommendations, selections for any occasion and new music";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

File: pkgs/by-name/ya/yandex-music/package.nix [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    homepage = "https://music.yandex.ru/";
+    downloadPage = "https://music.yandex.ru/download/";
+    changelog = "https://github.com/cucumber-sp/yandex-music-linux/releases/tag/v5.13.2";
```
Comment (dotlambda) [inline pkgs/by-name/ya/yandex-music/package.nix:83]:
```suggestion
    changelog = "https://github.com/cucumber-sp/yandex-music-linux/releases/tag/v${version}";
```

**Pull request merged**